habitat_baselines/common/tensor_dict.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import copy
+from typing import Callable, Dict, Optional, Tuple, Union, overload
+
+import numpy as np
+import torch
+
+TensorLike = Union[
+    torch.Tensor, np.ndarray, int, float, np.float32, np.float64
+]
+DictTree = Dict[str, Union[TensorLike, "DictTree"]]
+TensorIndexType = Union[int, slice, Tuple[Union[int, slice], ...]]
+
+
+def _to_tensor(v: TensorLike) -> torch.Tensor:
+    if torch.is_tensor(v):
+        return v
+    elif isinstance(v, np.ndarray):
+        return torch.from_numpy(v)
+    else:
+        return torch.tensor(v, dtype=torch.float)
+
+
+class TensorDict(dict):
```

Comment:
So extending dict lie this is usually not a great idea. Is there any reason ModuleDict, ParamterDict or https://pypi.org/project/dotmap/ or https://github.com/cdgriffith/Box/tree/master/box or something else? There are a lot of subtleties that are very easy to get wrong with Dot Dictionaries.

The set functions should have an explicit None return type btw.